lib/item/download_items_helper.rb

```diff
@@ -480,6 +483,17 @@ def generate_json_log(valids, invalids)
       json_log.to_json
     end
 
+    def filter_item_files(filenames, regexp_filter=DEFAULT_DOCUMENT_FILTER)
```

Comment:
What kind of unit testing would be needed? It would be expected that a valid Regexp object is being passed to the function, in addition to a syntactically correct array of filename strings. This method then just returns whichever of those files whose basename has some match against the Regexp.
Indirect testing with actual (valid and invalid) regular expressions happens in features/api.feature:638-675.